study_group/models.py

```diff
@@ -0,0 +1,49 @@
+from django.db import models
+from django.db.models import Q
+
+
+class GroupManager(models.Manager):
+    def get_all_non_full_groups(self):
+        return self.annotate(members_count=models.Count("group_members"))\
+            .filter(members_count__lt=models.F('capacity'))
+
+    def search_group_by_keyword(self, keyword: str):
+        return self.filter(Q(group_description__icontains=keyword) | Q(field__icontains=keyword))
+
+
+class StudyGroup(models.Model):
+    study_group = models.AutoField(primary_key=True)
+#    owner_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    field = models.CharField(max_length=30)
+    group_description = models.TextField()
+    capacity = models.PositiveIntegerField()
+    objects = GroupManager()
+
+    def join_group(self, user_id: int):
+        group_member = GroupMember(group_id=self, private_id=user_id)
+        group_member.save()
+
+    def is_group_full(self):
+        return self.group_members.count() >= self.capacity
+
+    def get_all_group_members(self):
+        return self.group_members.all()
+
+    # create a study group and add creator to group
+    @classmethod
+    def create(cls, **kwargs):
+        owner_id = kwargs.pop('owner_id', None)
+        # also need to check if owner_id exists in user table
+        if owner_id is None:
+            raise ValueError("Group owner doesn't exist")
```

Comment:
Doesn't this part of the user relationship?

Reply:
I figured its best to enforce adding owner to group in the creation method, given the owner ID. 
Waiting for the User table before I can implement it properly